packages/sdk/src/sdk/api/tracks/TracksApi.ts

```diff
@@ -405,12 +406,23 @@ export class TracksApi extends GeneratedTracksApi {
       'downloadTrack',
       RecordTrackDownloadSchema
     )(params)
+    const location = await getLocation({ logger: this.logger })
     return await this.entityManager.manageEntity({
       userId,
       entityType: EntityType.TRACK,
       entityId: trackId,
       action: Action.DOWNLOAD,
       auth: this.auth,
+      metadata: location
+        ? JSON.stringify({
+            cid: '',
+            data: {
+              city: location.city,
+              region: location.region,
+              country: location.country
```

Comment:
nit: could spread the location here `...location`

Reply:
because `data` here is free-form, i wanted the typing to be explicit, so i had intentionally not done this